crates/oxc_parser/src/lexer/number.rs

```diff
@@ -96,3 +96,42 @@ pub fn parse_big_int(s: &str, kind: Kind) -> Result<BigInt, &'static str> {
     };
     BigInt::parse_bytes(s.as_bytes(), radix).ok_or("invalid bigint")
 }
+
+#[inline]
```

Comment:
This is a deoptimization I think, it will bust the cpu cache for loading so many CPU instructions, `#[cold]` should be the better option here.

Remember, data oriented design, is thinking about the statistics.

Reply:
Sorry to contradict you Boshen, but I don't think `#[cold]` is the right option here either.

`#[cold]` is for telling the branch predictor "this function is rarely called", and so to assume another arm in a `match` is more likely to be the one which executes (or same in an `if`/`else` branch). But in this case, `without_underscores` is always called, so `#[cold]` won't do anything.

If want to make sure the function doesn't get inlined, use `#[inline(never)]`. Or just don't use any attribute, and let the compiler decide itself whether to inline or not. The latter is usually the best option.

FYI, I think I've over-used `#[cold]` in the lexer, and could probably remove some of them.